c_emulator/riscv_callbacks_if.h

```diff
@@ -2,64 +2,80 @@
 
 #include "sail.h"
 
-// This class is not declared in "riscv_callbacks.h" as that file is
-// included in the C model generated by Sail.
+namespace model {
+class Model;
+}
 
 class callbacks_if {
 public:
   virtual ~callbacks_if() = default;
-  // TODO: finding ways to improve the format
-  virtual void fetch_callback([[maybe_unused]] sbits opcode) { }
-  virtual void mem_write_callback([[maybe_unused]] const char *type,
+
+  virtual void fetch_callback([[maybe_unused]] model::Model &model,
```

Comment:
Is the namespace and class hardcoded in sail? I'd find riscv::Model slightly more readable. But no need to change this just an observation.

Reply:
Done